src/Sylius/Behat/Context/Api/Shop/ProductContext.php

```diff
@@ -662,4 +676,31 @@ private function hasProductWithMainImage(): bool
 
         return $images[0]['type'] === 'main' && $images[0]['path'];
     }
+
+    private function productHasAssociationsWithProducts(array $associationsIris, string $productAssociationName, array $products): bool
+    {
+        foreach ($associationsIris as $associationIri) {
+            $associationResponse = $this->client->showByIri($associationIri);
+            $associationTypeIri = $this->responseChecker->getValue($associationResponse, 'type');
+            $associationTypeResponse = $this->client->showByIri($associationTypeIri);
+
+            if ($this->responseChecker->hasValue($associationTypeResponse, 'name', $productAssociationName)) {
+               foreach ($products as $product) {
+                  $associatedProducts = $this->responseChecker->getValue($associationResponse, 'associatedProducts');
+                    if (!$this->productIsAssociated($product, $associatedProducts)) {
+                        return false;
+                    }
+               }
+            }
+        }
+
+        return true;
```

Comment:
It seems wrong. If I see correctly, the lack of requested association is directing us to true, which is not true - the product does not have another one within the association named X, because the association named X does not exist

```suggestion
            if ($this->responseChecker->hasValue($associationTypeResponse, 'name', $productAssociationName)) {
               foreach ($products as $product) {
                  $associatedProducts = $this->responseChecker->getValue($associationResponse, 'associatedProducts');
                    if (!$this->productIsAssociated($product, $associatedProducts)) {
                        return false;
                    }
               }
               
               return true;
            }
        }

        return false;
```